src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1582,8 +1599,17 @@ internal void ErrorReadNotifyUser(string? data)
             DataReceivedEventHandler? errorDataReceived = ErrorDataReceived;
             if (errorDataReceived != null)
             {
+                // Call back to user informing data is available.
                 DataReceivedEventArgs e = new DataReceivedEventArgs(data);
-                errorDataReceived(this, e); // Call back to user informing data is available.
+                ISynchronizeInvoke? syncObj = SynchronizingObject;
+                if (syncObj != null && syncObj.InvokeRequired)
```

Comment:
```suggestion
```suggestion
                if (SynchronizingObject is ISynchronizeInvoke syncObj && syncObj.InvokeRequired)
```
```